sql/expression/function/field.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+// Field joins several strings together.
+type Field struct {
+	args []sql.Expression
+}
+
+var _ sql.FunctionExpression = (*Field)(nil)
+var _ sql.CollationCoercible = (*Field)(nil)
+
+// NewField creates a new Field UDF.
+func NewField(args ...sql.Expression) (sql.Expression, error) {
+	if len(args) < 2 {
+		return nil, sql.ErrInvalidArgumentNumber.New("FIELD", "2 or more", len(args))
+	}
+
+	return &Field{args}, nil
+}
+
+// FunctionName implements sql.FunctionExpression
+func (f *Field) FunctionName() string {
+	return "field"
+}
+
+// Description implements sql.FunctionExpression
+func (f *Field) Description() string {
+	return "returns the string at index number."
+}
+
+// Type implements the Expression interface.
+func (f *Field) Type() sql.Type {
+	return types.Int32
```

Comment:
mysql reports this as a long when I'm testing

Reply:
This is what I get when running on MySQL with `--column-type-info`
```sql
mysql> select field('a', 'a');
Field 1
Name:      `field('a', 'a')`
Org_name:  ``
Catalog:   `def`
Database:  ``
Table:     ``
Org_table: ``
Type:      Integer
DbType:    TINY
Collation: binary (63)
Length:    3
Decimals:  0
Flags:     NOT_NULL NUM

+-----------------+
| field('a', 'a') |
+-----------------+
|               1 |
+-----------------+
1 row in set (0.0003 sec)
```